### PR TITLE
617 test services card view

### LINKED ISSUE #617: Services Gateway: unit test: services card view

### PR DESCRIPTION
**Pull request by 7ruth**
Resolves #617 

## Proposed Changes

  -Tests for ServicesCardView component
  
## How to Test

  -Run npm test

### CODE REVIEW

File: src/components/Main/scenes/FabricView/components/FabricGrid/components/FabricMainView/scene/ServicesCardsView/ServicesCardsView.test.js [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    expect(wrapper.find(GMServiceCardView).length).toBe(3);
+  });
+});
+
+describe("ServicesCardsView functionality", () => {
+  test("Has appropriate headings when group filter is applied ", () => {
+    const filteredServices = filterServicesByStatus("stable");
+    wrapper = mount(RouterWrap(["/"], filteredServices, "Owner"));
+
+    expect(wrapper.html().includes("Remote")).toBe(true);
```
Comment (karinakeen) [inline src/components/Main/scenes/FabricView/components/FabricGrid/components/FabricMainView/scene/ServicesCardsView/ServicesCardsView.test.js]: Maybe count the number of services being returned when group filter is applied.

File: src/components/Main/scenes/FabricView/components/FabricGrid/components/FabricMainView/scene/ServicesCardsView/ServicesCardsView.test.js [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    expect(wrapper.html().includes("AAC")).toBe(true);
+  });
+
+  test("Has only one GMServiceCardCollection when group filter is set to none", () => {
+    const filteredServices = filterServicesByStatus("stable");
+    wrapper = mount(RouterWrap(["/"], filteredServices, "None"));
+    expect(wrapper.find(GMServiceCardCollection).length).toBe(1);
+  });
+
+  test("Has a 1 GMServiceCardView when we only pass stable services to the component ", () => {
```
Comment (karinakeen) [inline src/components/Main/scenes/FabricView/components/FabricGrid/components/FabricMainView/scene/ServicesCardsView/ServicesCardsView.test.js]: you could add a test case for no services found

File: src/components/Main/scenes/FabricView/components/FabricGrid/components/FabricMainView/scene/ServicesCardsView/ServicesCardsView.test.js [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+    expect(wrapper.html().includes("Network")).toBe(true);
+    expect(wrapper.html().includes("Backroom")).toBe(true);
+    expect(wrapper.html().includes("Export")).toBe(true);
+    expect(wrapper.html().includes("Management")).toBe(true);
+    expect(wrapper.html().includes("Internet")).toBe(true);
+    expect(wrapper.html().includes("Domain")).toBe(true);
+    expect(wrapper.html().includes("Message")).toBe(true);
+    expect(wrapper.html().includes("Team")).toBe(true);
+    expect(wrapper.html().includes("User")).toBe(true);
+    expect(wrapper.html().includes("AAC")).toBe(true);
```
Comment (karinakeen) [inline src/components/Main/scenes/FabricView/components/FabricGrid/components/FabricMainView/scene/ServicesCardsView/ServicesCardsView.test.js]: looking for random-ish string could return false positive

Comment (7ruth) [reply src/components/Main/scenes/FabricView/components/FabricGrid/components/FabricMainView/scene/ServicesCardsView/ServicesCardsView.test.js]: Founds a better, way thank you for the tip^

File: src/components/Main/scenes/FabricView/components/FabricGrid/components/FabricMainView/scene/ServicesCardsView/ServicesCardsView.test.js (hunk 2 above)
Comment (7ruth) [reply src/components/Main/scenes/FabricView/components/FabricGrid/components/FabricMainView/scene/ServicesCardsView/ServicesCardsView.test.js]: Added..

File: src/components/Main/scenes/FabricView/components/FabricGrid/components/FabricMainView/scene/ServicesCardsView/ServicesCardsView.test.js (hunk 1 above)
Comment (7ruth) [reply src/components/Main/scenes/FabricView/components/FabricGrid/components/FabricMainView/scene/ServicesCardsView/ServicesCardsView.test.js]: Number of services will always be constant, but I found a better way to check heading.

File: src/components/Main/scenes/FabricView/components/FabricGrid/components/FabricMainView/scene/ServicesCardsView/ServicesCardsView.test.js [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+  });
+
+  test("Has a 3 GMServiceCardView when we pass services with all statuses ", () => {
+    expect(wrapper.find(GMServiceCardView).length).toBe(3);
+  });
+});
+
+describe("ServicesCardsView functionality", () => {
+  test("Has appropriate heading when group filter is applied ", () => {
+    const filteredServices = filterServicesByStatus("stable");
```
Comment (karinakeen) [inline src/components/Main/scenes/FabricView/components/FabricGrid/components/FabricMainView/scene/ServicesCardsView/ServicesCardsView.test.js]: please delete unused variables at the top and add a snapshot test

Comment (karinakeen) [reply src/components/Main/scenes/FabricView/components/FabricGrid/components/FabricMainView/scene/ServicesCardsView/ServicesCardsView.test.js]: you should add a test for cardsView function

File: src/components/Main/scenes/FabricView/components/FabricGrid/components/FabricMainView/scene/ServicesCardsView/ServicesCardsView.test.js [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+    wrapper = mount(RouterWrap(["/"], filteredServices, "Owner"));
+    expect(wrapper.find(GMServiceHeader).props().headerTitle).toBe("stable");
+    expect(wrapper.find(GMServiceHeader).props().showStatusIcon).toBe(false);
+  });
+
+  test("Has only one GMServiceCardCollection when group filter is set to none", () => {
+    const filteredServices = filterServicesByStatus("stable");
```
Comment (karinakeen) [inline src/components/Main/scenes/FabricView/components/FabricGrid/components/FabricMainView/scene/ServicesCardsView/ServicesCardsView.test.js]: if you are declaring variable filteredServices each time, might as well rename them as filteredServicesByStable, filteredServicesByNone, etc.

**Comment by karinakeen** [comment]
Looks good

**Pull request merged**